src/libaktualizr/uptane/secondaryinterface.h

```diff
@@ -20,21 +20,32 @@ namespace Uptane {
 
 class SecondaryInterface {
  public:
+  // This ctor should be removed as the secondary configuration SecondaryConfig
+  // is the secondaries's specific, see SecondaryConfig declaration
```

Comment:
You are probably right. I think a lot of this was done to make `SecondaryFactory` work smoothly, and since we are removing that, we can get rid of some of the associated cruft along with it.

Reply:
OK. I can try it in the next PRs since it leads to to many changes across the source code.